src/Symfony/Component/Cache/Adapter/AbstractAdapter.php

```diff
@@ -127,14 +131,19 @@ public static function createSystemCache($namespace, $defaultLifetime, $version,
 
     public static function createConnection($dsn, array $options = array())
     {
-        if (!is_string($dsn)) {
-            throw new InvalidArgumentException(sprintf('The %s() method expect argument #1 to be string, %s given.', __METHOD__, gettype($dsn)));
-        }
-        if (0 === strpos($dsn, 'redis://')) {
-            return RedisAdapter::createConnection($dsn, $options);
+        @trigger_error(sprintf('The %s() method is deprecated since version 3.4 and will be removed in 4.0. Use the DsnFactory::createConnection() method from Dsn component instead.', __METHOD__), E_USER_DEPRECATED);
+
+        try {
+            $type = ConnectionFactory::getType($dsn);
+        } catch (DsnInvalidArgumentException $e) {
+            throw new InvalidArgumentException($e->getMessage(), 0, $e);
```

Comment:
Just asking: if in the `catch` we just throw an exception ... would it better to remove the `catch` entirely to throw the original and more specific `DsnInvalidArgumentException `?

Reply:
👍 for throwing `DsnInvalidArgumentException`, as that already is a framework specific exception

Reply:
Wouldn't it be a BC break?

Reply:
it would yes, so the rethrow is required actually